src/mlpack/methods/ann/rbm/spike_slab_rbm_impl.hpp

```diff
@@ -119,6 +119,9 @@ RBM<InitializationRuleType, DataType, PolicyType>::Phase(
 
   visiblePenaltyGrad = -0.5 * arma::dot(input, input)
       / std::pow(input.n_cols, 2);
+
+  // To avoid 'unused variable' warning.
+  (void) visiblePenaltyGrad;
```

Comment:
If it's unused, I'd suggest just plain removing `visiblePenaltyGrad`.  But this raises the question of why it's even computed in the first place.  However, I don't know anything about `SpikeSlabRBM`, so I can't say. :(  @ShikharJ: any thoughts here?

Reply:
Yes, we can remove `visiblePenaltyGrad` by doing this:
```cpp
gradient(weightGrad.n_elem + spikeBiasGrad.n_elem) = -0.5 * arma::dot(input, input)
      / std::pow(input.n_cols, 2);
```

Reply:
It's merely an alias for a part of the `gradient` matrix. I'm fine with removing it, but it might reduce the readability.

Reply:
Ah, sorry, I did not recognize it was an alias.  I should have looked more closely. :)